### PR TITLE
Fix TypeError triggered by unexpected return from callable alias

### LINKED ISSUE #3598: Callable alias hung if returning list.
## xonfig

<details>

```
ftian@ftian-desktop:~$ xonsh
ftian@ftian-desktop ~ $ def _foo(args): 
.......................     return [1, 2], [3, 4] 
.......................                                                                         
ftian@ftian-desktop ~ $ aliases['foo'] = _foo                                                   
ftian@ftian-desktop ~ $ foo                                                                     
Exception in thread Thread-138:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.8/dist-packages/xonsh/__amalgam__.py", line 15040, in run
    self.returncode = parse_proxy_return(r, sp_stdout, sp_stderr)
  File "/usr/local/lib/python3.8/dist-packages/xonsh/__amalgam__.py", line 14785, in parse_proxy_return
    stdout.write(r[0])
TypeError: write() argument must be str, not list



```

</details>

## Expected Behavior

Error.

## Current Behavior

xonsh hung, ^C won't kill it either.

### Traceback (if applicable)

<details>

```
traceback
```

</details>

### PR DESCRIPTION
**Pull request by laloch**
Callable aliases returning tuple may raise `TypeError` exception effectively hanging the shell, if the tuple contains values of unexpected type.

Fixes #3598.

### CODE REVIEW

File: xonsh/proc.py [hunk 1]
```diff
@@ -1203,12 +1203,18 @@ def parse_proxy_return(r, stdout, stderr):
     elif isinstance(r, cabc.Sequence):
         rlen = len(r)
         if rlen > 0 and r[0] is not None:
-            stdout.write(r[0])
+            if isinstance(r[0], str):
+                stdout.write(r[0])
+            else:
+                stdout.write(str(r[0]))
```
Comment (scopatz) [inline xonsh/proc.py:1209]: I think this can just be `stdout.write(str(r[0]))`, because `str()` of a str object is is basically free. Same below

**Comment by scopatz** [comment]
Thanks for putting this in @laloch!

File: xonsh/proc.py (hunk 1 above)
Comment (laloch) [reply xonsh/proc.py:1209]: I always prefer explicit over implicit, so yes, this is intentional. But OK, I can change it.

Comment (laloch) [reply xonsh/proc.py]: Done.

**Pull request merged**